Core/Generators/CPlusPlus/CPlusPlusGenerator.cs

```diff
@@ -510,9 +516,9 @@ public override ValueTask<string> Compile(BebopSchema schema, GeneratorConfig co
                 }
             }
 
-            if (!string.IsNullOrWhiteSpace(Config.Namespace))
+            foreach (string ns in namespaces)
             {
-                builder.AppendLine($"}} // namespace {Config.Namespace}");
+                builder.AppendLine($"}} // namespace {ns}");
                 builder.AppendLine("");
             }
```

Comment:
Generally fine with this, but it would be better to handle your desired behavior behind an option, that way those who are targeting C++17 and above can get a single level namespace you can see an example of that here:
https://github.com/betwixt-labs/bebop/blob/accb3e961bcdb5b5fac0a41f46fdcc5a952fcdba/Core/Generators/CSharp/CSharpGenerator.cs#L29

I'll leave it to you to determine what an appropriate option name would be. For a refresher options are passed either via the bebop.json config or on the command line inside the generator string. ( "cpp:./GeneratedTestCode/test.g.hpp,namespace=Bebop::Codegen".

You'll need to test it works through both these means.

Reply:
Sorry I was busy with other things and only had free time during this holiday to revisit this. I think this should be using C++17 instead since bebop already relying on `std::variant`; which is only available on C++17 and newer